docs/src/api/class-locatorassertions.md

```diff
@@ -1018,6 +1033,122 @@ await Expect(
+
+Ensures the [Locator] points to an element with given CSS classes. All classes from the asserted value, separated by spaces, must be present in the [Element.classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) in any order.
+
+**Usage**
+
+```html
+<div class='middle selected row' id='component'></div>
+```
+
+```js
+const locator = page.locator('#component');
+await expect(locator).toContainClass('middle selected row');
+await expect(locator).toContainClass('selected');
+await expect(locator).toContainClass('row middle');
+```
+
+```java
+assertThat(page.locator("#component")).containsClass("middle selected row");
+assertThat(page.locator("#component")).containsClass("selected");
+assertThat(page.locator("#component")).containsClass("row middle");
+```
+
+```python async
+from playwright.async_api import expect
+
+locator = page.locator("#component")
+await expect(locator).to_contain_class("middle selected row")
+await expect(locator).to_contain_class("selected")
+await expect(locator).to_contain_class("row middle")
+```
+
+```python sync
+from playwright.sync_api import expect
+
+locator = page.locator("#component")
+expect(locator).to_contain_class("middle selected row")
+expect(locator).to_contain_class("selected")
+expect(locator).to_contain_class("row middle")
+```
+
+```csharp
+var locator = Page.Locator("#component");
+await Expect(locator).ToContainClassAsync("middle selected row");
+await Expect(locator).ToContainClassAsync("selected");
+await Expect(locator).ToContainClassAsync("row middle");
+```
+
+When an array is passed, the method asserts that the list of elements located matches the corresponding list of expected class lists. Each element's class attribute is matched against the corresponding class in the array:
+
+```html
+<div class='list'></div>
+  <div class='component inactive'></div>
+  <div class='component active'></div>
+  <div class='component inactive'></div>
+</div>
+```
+
+```js
+const locator = page.locator('list > .component');
+await expect(locator).toContainClass(['inactive', 'active', 'inactive']);
+```
+
+```java
+assertThat(page.locator("list > .component")).containsClass(new String[] {"inactive", "active", "inactive"});
+```
+
+```python async
+from playwright.async_api import expect
+
+locator = page.locator("list > .component")
+await expect(locator).to_contain_class(["inactive", "active", "inactive"])
+```
+
+```python sync
+from playwright.sync_api import expect
+
+locator = page.locator("list > .component")
+await expect(locator).to_contain_class(["inactive", "active", "inactive"])
+```
+
+```csharp
+var locator = Page.Locator("list > .component");
+await Expect(locator).ToContainClassAsync(new string[]{"inactive", "active", "inactive"});
+```
+
+### param: LocatorAssertions.toContainClass.expected
+* since: v1.52
+* langs: js
+- `expected` <[string]|[Array]<[string]>>
+
+Expected class or RegExp or a list of those.
+
+### param: LocatorAssertions.toContainClass.expected
```

Comment:
Once you remove RegExp from here, it seems to be exactly the same as JS, so let's merge.